compiler/test/runnable/test23710.d

```diff
@@ -4,13 +4,14 @@ int test(int i)
 {
     int j;
     int[] foo;
+    int[] bar = [1,2,3];
     if (0)
     {
         for (;;)
         {
             import core.stdc.stdio;
             printf("start body\n");
-            foo = foo ~ [1,2,3];
+            foo = foo ~ bar;
```

Comment:
Why is this change needed?

Reply:
https://github.com/dlang/dmd/pull/14550#discussion_r1166609165

Reply:
This is valid code, so if this PR makes it miscompile or error, that's bad. Discovering such bugs in the compiler is why there is a testsuite :-) So fix the PR, not the test! ;-)